dotnet/samples/Concepts/Filtering/TelemetryWithFilters.cs

```diff
@@ -0,0 +1,153 @@
+    {
+        // Initialize kernel with chat completion service.
+        var builder = Kernel
+            .CreateBuilder()
+            .AddOpenAIChatCompletion("gpt-4", TestConfiguration.OpenAI.ApiKey);
+
+        // Create and add logger, which will output messages to test detail summary window.
+        var logger = this.LoggerFactory.CreateLogger<TelemetryWithFilters>();
+        builder.Services.AddSingleton<ILogger>(logger);
+
+        // Add filters with logging.
+        builder.Services.AddSingleton<IFunctionInvocationFilter, FunctionInvocationLoggingFilter>();
+        builder.Services.AddSingleton<IPromptRenderFilter, PromptRenderLoggingFilter>();
+        builder.Services.AddSingleton<IAutoFunctionInvocationFilter, AutoFunctionInvocationLoggingFilter>();
+
+        var kernel = builder.Build();
+
+        // Import sample functions.
+        kernel.ImportPluginFromFunctions("HelperFunctions",
+        [
+            kernel.CreateFunctionFromMethod(() => DateTime.UtcNow.ToString("R"), "GetCurrentUtcTime", "Retrieves the current time in UTC."),
+            kernel.CreateFunctionFromMethod((string cityName) =>
+                cityName switch
+                {
+                    "Boston" => "61 and rainy",
+                    "London" => "55 and cloudy",
+                    "Miami" => "80 and sunny",
+                    "Paris" => "60 and rainy",
+                    "Tokyo" => "50 and sunny",
+                    "Sydney" => "75 and sunny",
+                    "Tel Aviv" => "80 and sunny",
+                    _ => "31 and snowing",
+                }, "GetWeatherForCity", "Gets the current weather for the specified city"),
+        ]);
+
+        // Enable automatic function calling.
+        var executionSettings = new OpenAIPromptExecutionSettings { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+
+        // Invoke prompt with arguments.
+        const string Prompt = "Given the current time of day and weather, what is the likely color of the sky in {{$city}}?";
+        var result = await kernel.InvokePromptAsync(Prompt, new(executionSettings) { ["city"] = "Boston" });
+
+        Console.WriteLine(result);
+    }
+
+    /// <summary>
+    /// Filter which logs an information available during function invocation such as:
+    /// Function name, arguments, execution settings, result, duration, token usage.
+    /// </summary>
+    private sealed class FunctionInvocationLoggingFilter(ILogger logger) : IFunctionInvocationFilter
+    {
+        public async Task OnFunctionInvocationAsync(FunctionInvocationContext context, Func<FunctionInvocationContext, Task> next)
+        {
+            long startingTimestamp = Stopwatch.GetTimestamp();
+
+            logger.LogInformation("Function {FunctionName} invoking.", context.Function.Name);
+
+            if (context.Arguments.Count > 0)
+            {
+                logger.LogTrace("Function arguments: {Arguments}", context.Arguments);
```

Comment:
Does arguments also need to be serialized?

Reply:
By default, `IDictionary` is handled by logger correctly, that's why I don't see any issues in output, I also tested this behavior in empty console application:
```csharp
var dictionary = new Dictionary<string, string> { ["key"] = "value" };

logger.LogInformation("Test: {Dictionary}", dictionary);
```

Output:
```
info: ConsoleApp1.Program[0]
      Test: [key, value]
```

But I see in SK telemetry we actually serialize it, so maybe it will be better to do that in example as well, I will make the update:
https://github.com/microsoft/semantic-kernel/blob/9ac1152a1f5cb402e623ca4bea0b3f6019a7d47f/dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionLogMessages.cs#L45

Reply:
Updated with arguments serialization.